src/frontend/src/pages/HomePage/UpcomingDeadlines.tsx

```diff
@@ -42,61 +48,59 @@ const UpcomingDeadlines: React.FC = () => {
         justifyContent: 'flex-start'
       }}
     >
-      {workPackages.data?.length === 0
-        ? 'No upcoming deadlines'
-        : workPackages.data?.map((wp) => (
-            <Card
-              variant="outlined"
-              key={wbsPipe(wp.wbsNum)}
-              sx={{ minWidth: 'fit-content', mr: 3, background: theme.palette.background.default }}
-            >
-              <CardContent sx={{ padding: 3 }}>
-                <Link
-                  variant="h6"
-                  component={RouterLink}
-                  to={`${routes.PROJECTS}/${wbsPipe(wp.wbsNum)}`}
-                  sx={{ marginBottom: 2 }}
-                >
-                  {wbsPipe(wp.wbsNum)} - {wp.name}
-                </Link>
+      {workPackages.data?.length === 0 ? (
+        <Typography>No upcoming deadlines</Typography>
+      ) : (
+        workPackages.data?.map((wp) => (
+          <Card
+            variant="outlined"
+            key={wbsPipe(wp.wbsNum)}
+            sx={{ minWidth: 'fit-content', mr: 3, background: theme.palette.background.default }}
+          >
+            <CardContent sx={{ padding: 3 }}>
+              <Link
+                variant="h6"
+                component={RouterLink}
+                to={`${routes.PROJECTS}/${wbsPipe(wp.wbsNum)}`}
+                sx={{ marginBottom: 2 }}
+              >
+                {wbsPipe(wp.wbsNum)} - {wp.name}
+              </Link>
+              <Box>
+                <Box>
+                  <UpcomingDeadlinesDetailDisplay
+                    label="End Date"
+                    content={datePipe(wp.endDate)}
+                  ></UpcomingDeadlinesDetailDisplay>
```

Comment:
`<div ...></div>` === `<div ... />` and you should fix this everywhere you have it

Reply:
actually its ok replace all blessed it